### ISSUE TITLE
local variable 'o' referenced before assignment

### ISSUE THREAD

**Issue by cuigh** [kijai/ComfyUI-CogVideoXWrapper#143]
An error occurred after installing SageAttention. 4090 on Linux.

Workflow: https://github.com/kijai/ComfyUI-CogVideoXWrapper/blob/main/examples/cogvideox_I2V_example_01.json

```
!!! Exception during processing !!! local variable 'o' referenced before assignment
Traceback (most recent call last):
  File "/data/users/cuigh/ComfyUI/execution.py", line 323, in execute
    output_data, output_ui, has_subgraph = get_output_data(obj, input_data_all, execution_block_cb=execution_block_cb, pre_execute_cb=pre_execute_cb)
  File "/data/users/cuigh/ComfyUI/execution.py", line 198, in get_output_data
    return_values = _map_node_over_list(obj, input_data_all, obj.FUNCTION, allow_interrupt=True, execution_block_cb=execution_block_cb, pre_execute_cb=pre_execute_cb)
  File "/data/users/cuigh/ComfyUI/execution.py", line 169, in _map_node_over_list
    process_inputs(input_dict, i)
  File "/data/users/cuigh/ComfyUI/execution.py", line 158, in process_inputs
    results.append(getattr(obj, func)(**inputs))
  File "/data/users/cuigh/ComfyUI/custom_nodes/ComfyUI-CogVideoXWrapper/nodes.py", line 953, in process
    latents = pipeline["pipe"](
  File "/home/ubuntu/.conda/envs/comfyui/lib/python3.10/site-packages/torch/utils/_contextlib.py", line 116, in decorate_context
    return func(*args, **kwargs)
  File "/data/users/cuigh/ComfyUI/custom_nodes/ComfyUI-CogVideoXWrapper/pipeline_cogvideox.py", line 835, in __call__
    noise_pred = self.transformer(
  File "/home/ubuntu/.conda/envs/comfyui/lib/python3.10/site-packages/torch/nn/modules/module.py", line 1553, in _wrapped_call_impl
    return self._call_impl(*args, **kwargs)
  File "/home/ubuntu/.conda/envs/comfyui/lib/python3.10/site-packages/torch/nn/modules/module.py", line 1562, in _call_impl
    return forward_call(*args, **kwargs)
  File "/data/users/cuigh/ComfyUI/custom_nodes/ComfyUI-CogVideoXWrapper/custom_cogvideox_transformer_3d.py", line 615, in forward
    hidden_states, encoder_hidden_states = block(
  File "/home/ubuntu/.conda/envs/comfyui/lib/python3.10/site-packages/torch/nn/modules/module.py", line 1553, in _wrapped_call_impl
    return self._call_impl(*args, **kwargs)
  File "/home/ubuntu/.conda/envs/comfyui/lib/python3.10/site-packages/torch/nn/modules/module.py", line 1562, in _call_impl
    return forward_call(*args, **kwargs)
  File "/data/users/cuigh/ComfyUI/custom_nodes/ComfyUI-CogVideoXWrapper/custom_cogvideox_transformer_3d.py", line 288, in forward
    attn_hidden_states, attn_encoder_hidden_states = self.attn1(
  File "/home/ubuntu/.conda/envs/comfyui/lib/python3.10/site-packages/torch/nn/modules/module.py", line 1553, in _wrapped_call_impl
    return self._call_impl(*args, **kwargs)
  File "/home/ubuntu/.conda/envs/comfyui/lib/python3.10/site-packages/torch/nn/modules/module.py", line 1562, in _call_impl
    return forward_call(*args, **kwargs)
  File "/home/ubuntu/.conda/envs/comfyui/lib/python3.10/site-packages/diffusers/models/attention_processor.py", line 490, in forward
    return self.processor(
  File "/data/users/cuigh/ComfyUI/custom_nodes/ComfyUI-CogVideoXWrapper/custom_cogvideox_transformer_3d.py", line 100, in __call__
    hidden_states = sageattn(query, key, value, is_causal=False)
  File "/home/ubuntu/.conda/envs/comfyui/lib/python3.10/site-packages/sageattention/core.py", line 87, in sageattn
    return o
UnboundLocalError: local variable 'o' referenced before assignment
```

**Comment by kijai** [comment]
Sageattn update broke it for some reason, to me it looks like bug in their code, simplest fix would be to edit the core.py of the sageattn code so that the line `dtype = q.dtype` would instead be `dtype = v.dtype`

I'll ask them if this was intentional as I don't see how to make it work otherwise.

**Comment by kijai** [comment]
This should now be fixed in sageattn 1.0.2

**Comment by cuigh** [comment]
Confirmed that v1.0.2 has fixed this problem, thx.
